packages/editor/src/components/StatesTable/StateEditor.react.js

```diff
@@ -13,19 +13,22 @@ import withConfirmDialog from '../ConfirmDialog';
 import ErrorLabel from '../ErrorLabel.react';
 
 @withConfirmDialog
-export default class StatesEditor extends PureComponent {
+export default class StateEditor extends PureComponent {
   static propTypes = {
-    state: statePropTypes,
+    fsmState: statePropTypes,
     onClose: PropTypes.func.isRequired,
     onSave: PropTypes.func.isRequired,
-    existingStates: PropTypes.arrayOf(PropTypes.string).isRequired
+    existingStates: PropTypes.arrayOf(PropTypes.string).isRequired,
+    stateConfig: PropTypes.shape({
```

Comment:
- I would use and 'availableNames' for all possible passing state names -> e.g. show select box instead of input